### PR TITLE
feat(tabs): removable tabs

### PR DESCRIPTION
**Pull request by vvpanchenko**
Enlarge Tabs component with possibility to remove tabs.
Demo page and Readme file, updated.

### CODE REVIEW

File: components/tabs/tabset.component.ts [hunk 1]
```diff
@@ -84,12 +87,41 @@ export class Tabset implements OnInit {
...
     }
 
-    this.tabs.slice(index, 1);
+    tab.removed.emit(tab);
+    this.tabs.splice(index, 1);
+  }
+
+  private getClosestTabIndex (index:number):number {
+    let tabsLength = this.tabs.length;
+    if (tabsLength) {
```
Comment (valorkin) [inline components/tabs/tabset.component.ts]: please invert logic

File: components/tabs/tabset.component.ts [hunk 2]
```diff
@@ -84,12 +87,41 @@ export class Tabset implements OnInit {
...
+          return nextIndex;
+        }
+      }
+    }
+    return -1;
+  }
+
+  private hasAvailableTabs (index:number) {
+    var tabsLength = this.tabs.length;
+    if (tabsLength) {
```
Comment (valorkin) [inline components/tabs/tabset.component.ts]: please invert logic

File: components/tabs/tabset.component.ts (hunk 1 above)
Comment (vvpanchenko) [inline components/tabs/tabset.component.ts]: Logic was not changed, meaning:
1. check "active", only if True, call "hasAvailableTabs".
2. hasAvailableTabs, at first check tabs length as it was.
3. Additional logic is to check if there is any sense to search for Tab that could be switched to active.
4. And finally only if all negative cases skipped, call "getClosestTabIndex", that will start to search in both directions left/right from current index.

So, previous logic was saved, but enlarged, is it OK ?

Comment (valorkin) [inline components/tabs/tabset.component.ts]:
```
if (!tabsLength) {
  return -1;
}
...
```

File: components/tabs/tabset.component.ts (hunk 2 above)
Comment (valorkin) [inline components/tabs/tabset.component.ts]:
```
if (!tabsLength) {
  return false;
}
...
```

File: components/tabs/tabset.component.ts (hunk 1 above)
Comment (vvpanchenko) [inline components/tabs/tabset.component.ts]: Yes, but that case still exists in the same place
Check please, source of function [removeTab](https://github.com/vvpanchenko/ng2-bootstrap/blob/master/components/tabs/tabset.component.ts#L84)

**Comment by vvpanchenko** [comment]
Updated request with new commit, check please

**Comment by valorkin** [comment]
Looks good, please squash commits and force push to your local branch

**Comment by vvpanchenko** [comment]
Done, Thanx for your time.

**Comment by valorkin** [comment]
LGTM!

**Pull request merged**